src/containerapp/azext_containerapp/custom.py

```
@@ -2370,6 +2374,57 @@ def delete_environment_telemetry_data_dog(cmd,
+    )
+    
+    containerapp_env_def = None
+    try:
+        containerapp_env_def = containerapp_env_telemetry_data_dog_decorator.show()
+    except Exception as e:
+        handle_raw_exception(e)
+
+    if not containerapp_env_def:
+        raise ResourceNotFoundError("The containerapp environment '{}' does not exist".format(name))
+
+    try:
+        r = containerapp_env_def["properties"]["openTelemetryConfiguration"]["destinationsConfiguration"]["dataDogConfiguration"]
+        if r is None:
+            raise ValidationError("The containerapp environment '{}' does not have data dog enabled.".format(name)) from e
+        r = {}
+
+        if "key" in containerapp_env_def["properties"]["openTelemetryConfiguration"]["destinationsConfiguration"]["dataDogConfiguration"]:
+            safe_set(r, "dataDogConfiguration", "key", value=DEFAULT_CONFIGURED_STR)
+
+        safe_set(r, "dataDogConfiguration", "site", value=safe_get(containerapp_env_def, "properties", "openTelemetryConfiguration", "destinationsConfiguration", "dataDogConfiguration", "site"))
+
+        enable_open_telemetry_traces = False
+        existing_traces = safe_get(containerapp_env_def, "properties", "openTelemetryConfiguration", "tracesConfiguration", "destinations")
+        if existing_traces and DATA_DOG_DEST in existing_traces:
+            enable_open_telemetry_traces = True
+        
+        safe_set(r, "dataDogConfiguration", "enable-open-telemetry-traces", value=enable_open_telemetry_traces)
+
+        enable_open_telemetry_metrics = False
+        existing_metrics = safe_get(containerapp_env_def, "properties", "openTelemetryConfiguration", "metricsConfiguration", "destinations")
+        if existing_metrics and DATA_DOG_DEST in existing_metrics:
+            enable_open_telemetry_metrics = True
+        
+        safe_set(r, "dataDogConfiguration", "enable-open-telemetry-metrics", value=enable_open_telemetry_metrics)
+
+        return r
+    except Exception as e:
```

is is necessary to catch all exception?